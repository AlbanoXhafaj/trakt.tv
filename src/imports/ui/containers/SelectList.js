import React from "react";
import PropTypes from 'prop-types';

//DropdownList generate by an array of strings
//title - describes the dropwdownlist user
//options - array of strings, elements of the list
//onChange - function that will accept as a parametre the index of the element
//fistElement - the first element of the list like "select options below"

const SelectList = ({ title, options , onChange , firstElement = null }) => (
  <div className="inline">
    <span> { title }: </span>
    <select onChange = { (e) => onChange( parseInt(e.target.value, 10) ) } >
      {
        firstElement
          ? <option value="-1"> { firstElement } </option>
          :null
      }
      {
        //create <option> tags with index as value
        options.map( (option,index) => (
          <option
            key= { index }
            value = { index }
          >
            { option }
          </option>
        ))
      }
    </select>
  </div>
);

SelectList.propTypes = {
  title: PropTypes.string.isRequired,
  options: PropTypes.array.isRequired,
  onChange: PropTypes.func.isRequired,
  fistElement: PropTypes.string
};

export default SelectList;
