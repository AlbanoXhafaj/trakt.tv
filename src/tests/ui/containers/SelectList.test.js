import "../../setups/setup_enzyme.js";
import React from "react";

import { shallow , mount } from "enzyme";
import toJSON from 'enzyme-to-json';

import SelectList from "containers/SelectList";

describe("<SelectList />" , () => {
  const onChange = jest.fn();
  const initialProps = [
    {
      title: "Dropdownlist without fisrt element",
      options: ["option1","option2","option3","option4","option5"],
      onChange: onChange
    },
    {
      title: "Dropdownlist with fisrt element",
      firstElement: "Select an option",
      options: ["option1","option2","option3","option4","option5"],
      onChange: onChange
    }
  ];

  let wrapper = shallow(<SelectList {...initialProps[0]} />);

  it("should match the snapshot without firstElement" ,() => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  })

  wrapper = shallow(<SelectList {...initialProps[1]} />);
  it("should match the snapshot with firstElement" ,() => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  })

  it("should pass the index as an argument on tha change event",()=> {
    const event = {
      preventDefault(){},
      target: { value: 3 }
    }

    wrapper.find("select").simulate('change', event);
    expect(onChange).toBeCalledWith(3);
  })
});
