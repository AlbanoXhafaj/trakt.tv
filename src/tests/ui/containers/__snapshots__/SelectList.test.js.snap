// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectList /> should match the snapshot with firstElement 1`] = `
<div
  className="inline"
>
  <span>
     
    Dropdownlist with fisrt element
    : 
  </span>
  <select
    onChange={[Function]}
  >
    <option
      value="-1"
    >
       
      Select an option
       
    </option>
    <option
      key="0"
      value={0}
    >
      option1
    </option>
    <option
      key="1"
      value={1}
    >
      option2
    </option>
    <option
      key="2"
      value={2}
    >
      option3
    </option>
    <option
      key="3"
      value={3}
    >
      option4
    </option>
    <option
      key="4"
      value={4}
    >
      option5
    </option>
  </select>
</div>
`;

exports[`<SelectList /> should match the snapshot without firstElement 1`] = `
<div
  className="inline"
>
  <span>
     
    Dropdownlist with fisrt element
    : 
  </span>
  <select
    onChange={[Function]}
  >
    <option
      value="-1"
    >
       
      Select an option
       
    </option>
    <option
      key="0"
      value={0}
    >
      option1
    </option>
    <option
      key="1"
      value={1}
    >
      option2
    </option>
    <option
      key="2"
      value={2}
    >
      option3
    </option>
    <option
      key="3"
      value={3}
    >
      option4
    </option>
    <option
      key="4"
      value={4}
    >
      option5
    </option>
  </select>
</div>
`;
